contracts/pcv/utils/ERC20Allocator.sol

```diff
@@ -0,0 +1,390 @@
+/// First application is allocating funds from a PSM to a yield venue so that liquid reserves are minimized.
+/// This contract should never hold PCV, however it has a sweep function, so if tokens get sent to it accidentally,
+/// they can still be recovered.
+/// @author Elliot Friedman
+contract ERC20Allocator is IERC20Allocator, CoreRef, RateLimitedV2 {
+    using Address for address payable;
+    using SafeERC20 for IERC20;
+    using SafeCast for *;
+
+    /// @notice container that stores information on all psm's and their respective deposits
+    struct depositInfo {
+        /// @notice target address to send excess tokens, will be a Compound PCV Deposit
+        address pcvDeposit;
+        /// @notice target token address to send
+        address token;
+        /// @notice only skim if balance of target is greater than targetBalance
+        /// only drip if balance of target is less than targetBalance
+        uint248 targetBalance;
+        /// @notice decimal normalizer to ensure buffer is updated uniformly across all deposits
+        int8 decimalsNormalizer;
+    }
+
+    /// @notice map the psm address to the corresponding deposit information
+    /// excess tokens past target balance will be pulled from the PSM
+    /// if PSM has less than the target balance,  will be pulled from the PSM
+    mapping(address => depositInfo) public allDeposits;
+
+    /// @notice ERC20 Allocator constructor
+    /// @param _core Volt Core for reference
+    /// @param _maxRateLimitPerSecond maximum rate limit per second that governance can set
+    /// @param _rateLimitPerSecond starting rate limit per second
+    /// @param _bufferCap cap on buffer size for this rate limited instance
+    constructor(
+        address _core,
+        uint256 _maxRateLimitPerSecond,
+        uint128 _rateLimitPerSecond,
+        uint128 _bufferCap
+    )
+        CoreRef(_core)
+        RateLimitedV2(_maxRateLimitPerSecond, _rateLimitPerSecond, _bufferCap)
+    {}
+
+    /// ----------- Governor Only API -----------
+
+    /// @notice create a new deposit
+    /// @param psm Peg Stability Module for this deposit
+    /// @param pcvDeposit that this PSM is linked to
+    /// @param targetBalance target amount of tokens for the PSM to hold
+    /// @param decimalsNormalizer decimal normalizer to ensure buffer is depleted and replenished properly
+    function createDeposit(
+        address psm,
+        address pcvDeposit,
+        uint248 targetBalance,
+        int8 decimalsNormalizer
+    ) external override onlyGovernor {
+        address token = PCVDeposit(pcvDeposit).balanceReportedIn();
+
+        require(
+            allDeposits[psm].token == address(0) &&
+                allDeposits[psm].pcvDeposit == address(0),
+            "ERC20Allocator: cannot overwrite existing deposit"
+        );
+        require(
+            token != address(0),
+            "ERC20Allocator: underlying token invalid"
+        );
```

Comment:
do we need this check? ultimately it could be any arbitrary address being passed

Reply:
we do. token is pulled from the pcv deposit and if address 0, it means the pcv deposit was not set up properly.